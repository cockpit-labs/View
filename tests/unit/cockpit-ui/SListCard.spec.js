import { shallowMount } from '@vue/test-utils'
import SListCard from '@/plugins/cockpit-ui/components/list-card/SListCard'

describe('List card', () => {
  const wrapper = shallowMount(SListCard, {
    slots: {
      title: 'Follow Up POS',
      body: 'Cum haec taliaque sollicitas eius aures everberarent expositas semper eius modi rumoribu',
      footer: '4 days left',
      'action-label': 'Start'
    }
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit click event on click action button', async () => {
    const button = wrapper.find('sbutton-stub')
    const spy = jest.fn()
    wrapper.vm.$on('click', spy)
    button.vm.$emit('click')
    await wrapper.vm.$nextTick()
    expect(spy).toHaveBeenCalledTimes(1)
  })
})
