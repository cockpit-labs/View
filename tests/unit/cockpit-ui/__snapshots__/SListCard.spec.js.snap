// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List card should match snapshot 1`] = `
<div
  class="s-list-card"
>
  <div
    class="title"
  >
    Follow Up POS
  </div>
   
  <div
    class="body"
  >
    Cum haec taliaque sollicitas eius aures everberarent expositas semper eius modi rumoribu
  </div>
   
  <!---->
   
  <div
    class="footer"
  >
    <div
      class="info"
    >
      4 days left
    </div>
     
    <sbutton-stub
      icon="angle-right"
      iconposition="right"
      rounded="true"
    >
      Start
    </sbutton-stub>
  </div>
</div>
`;
